compression/compression-bad-chunks.tentative.any.js

```diff
@@ -39,36 +39,67 @@ const badChunks = [
       return new Uint8Array(new WebAssembly.Memory({ shared:true, initial:1, maximum:1 }).buffer)
     }
   },
+  {
+    name: 'invalid deflate bytes',
+    value: new Uint8Array([0, 156, 75, 173, 40, 72, 77, 46, 73, 77, 81, 200, 47, 45, 41, 40, 45, 1, 0, 48, 173, 6, 36])
+  },
+  {
+    name: 'invalid gzip bytes',
+    value: new Uint8Array([0, 139, 8, 0, 0, 0, 0, 0, 0, 3, 75, 173, 40, 72, 77, 46, 73, 77, 81, 200, 47, 45, 41, 40, 45, 1, 0, 176, 1, 57, 179, 15, 0, 0, 0])
+  },
 ];
 
+// Test Case Design
+// We need to wait until after we close the writable stream to check if the decoded stream is valid.
+// We can end up in a state where all reads/writes are valid, but upon closing the writable stream an error is detected.
+// (Example: A zlib encoded chunk w/o the checksum).
+
 for (const chunk of badChunks) {
   promise_test(async t => {
-    const cs = new CompressionStream('gzip');
-    const reader = cs.readable.getReader();
-    const writer = cs.writable.getWriter();
-    const writePromise = writer.write(chunk.value);
-    const readPromise = reader.read();
-    await promise_rejects_js(t, TypeError, writePromise, 'write should reject');
-    await promise_rejects_js(t, TypeError, readPromise, 'read should reject');
+    const ds = new DecompressionStream('gzip');
```

Comment:
I'm confused. It looks like you're changing the compression tests to be decompression tests?

Reply:
You are right. I messed up here and copied my changes to the wrong file.

Thanks for catching this. Will have a new patch up this evening with the fix.

Reply:
Uploaded new patch with the correct file modified.